File: Binary Search Algorithm/Python/Binary-Search-Algorithm.py
def binary_search(arr, x):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            left = mid + 1
        else:
            right = mid - 1

    return -1

arr = [2, 3, 4, 10, 40]
x = 10

result = binary_search(arr, x)
if result != -1:
    print(f"Element found at index {result}")
else:
    print("Element not found in the array")

File: Binary Search Algorithm/Go/Binary-Search-Algorithm.go
package main

import "fmt"

func binarySearch(arr []int, x int) int {
    left := 0
    right := len(arr) - 1

    for left <= right {
        mid := left + (right - left) / 2

        if arr[mid] == x {
            return mid
        } else if arr[mid] < x {
            left = mid + 1
        } else {
            right = mid - 1
        }
    }

    return -1
}

func main() {
    arr := []int{2, 3, 4, 10, 40}
    x := 10

    result := binarySearch(arr, x)
    if result != -1 {
        fmt.Println("Element found at index", result)
    } else {
        fmt.Println("Element not found in the array")
    }
}
